src/main/groovy/org/labkey/gradle/plugin/TeamCity.groovy

```diff
@@ -301,73 +301,92 @@ class TeamCity extends Tomcat
         }
     }
 
-    private static void connect(AttachingConnector connector, int port) throws IllegalConnectorArgumentsException, IOException, IncompatibleThreadStateException
+    private static void threadDumpAndKill(AttachingConnector connector, int port) throws IllegalConnectorArgumentsException, IOException, IncompatibleThreadStateException
     {
-        Map<String, Connector.Argument> arguments = connector.defaultArguments();
-        arguments.get("hostname").setValue("localhost");
-        arguments.get("port").setValue(Integer.toString(port));
-        println("Attempting to shutdown Tomcat on debug port: " + port);
+        Map<String, Connector.Argument> arguments = connector.defaultArguments()
+        arguments.get("hostname").setValue("localhost")
+        arguments.get("port").setValue(Integer.toString(port))
+        long startTime = System.currentTimeMillis()
+        VirtualMachine vm
+
         try
         {
-            VirtualMachine vm = connector.attach(arguments);
-            vm.suspend();
-            for (ThreadReference threadReference : vm.allThreads())
+            vm = connector.attach(arguments)
+        }
+        catch (IOException ignore)
+        {
+            println("Unable to connect to VM at localhost:" + port + ", VM may already be shut down")
+            return
+        }
+
+        println("Waiting for graceful Tomcat shutdown.")
+        while (System.currentTimeMillis() - startTime < 15_000)
```

Comment:
Maybe create a constant for this (int seconds = 15) and reference it in the message below ("did not shutdown in __ seconds").